### PR TITLE
Add IDE versions to the dashboard

### PR DESCRIPTION
**Pull request by filiptronicek**
## Description
This PR is simply adding versions below the icons of all IDEs in the IDE switcher component (used during on-boarding and always available in the user preferences). 

This is partially a lie because we don't associate VS Code Desktop with a version, but it holds true for the rest.
image

It's a relatively easy win because we:
1. Already tag all newly-built IDE images with their version strings (https://github.com/gitpod-io/gitpod/pull/15082, https://github.com/gitpod-io/gitpod/pull/14055)
2. We already implemented the functionality in [`ide-service`](https://github.com/gitpod-io/gitpod/tree/main/components/ide-service) so that it returns the IDE versions along the IDE options (https://github.com/gitpod-io/gitpod/pull/15081)

With this we can do two things at once:
- keep our users informed what versions we use right from the Dashboard (good for checking new versions with bug fixes and features)
- makes it easier for us to check that our IDEs are at their correct versions when releasing them at IDE (the team).

## How to test

1. Go to the [preview environment](https://ft-add-idebd181cc38e.preview.gitpod-dev.com/workspaces) and sign in
2. You should be greeted with a modal with the IDEs and their versions, you can also visit the [user prefs](https://ft-add-idebd181cc38e.preview.gitpod-dev.com/preferences) to see it.

## Release Notes

```release-note
Added versions of all the suppoerted IDEs to the Preferences page
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

File: components/dashboard/src/settings/SelectIDE.tsx
Comment (akosyakov) [reply components/dashboard/src/settings/SelectIDE.tsx:81]: yes, it can be removed already cc @mustard-mh

**Comment by akosyakov** [comment]
@filiptronicek very nice job ⭐ 

@mustard-mh Could you test please it to bring it in, looking forward! 🚀

File: components/dashboard/src/settings/SelectIDE.tsx
Comment (akosyakov) [reply components/dashboard/src/settings/SelectIDE.tsx:183]: option has 2 versions stable and latest, passed version is a selection one of another depending on a user choice (line 98 above)

**Review by easyCZ — APPROVED** [review]
LGTM

/hold for you to merge when ready

**Review by gtsiolis — CHANGES_REQUESTED** [review]
Seems like this breaks the dashboard UX. ❗ 

| Modal (Signing up) | Preferences (User Settings) |
|-|-|
| Screenshot 2022-12-05 at 1 43 56 PM (2) | Screenshot 2022-12-05 at 1 45 10 PM (2) |

**Comment by gtsiolis** [comment]
Regarding the comment above in https://github.com/gitpod-io/gitpod/pull/15139#pullrequestreview-1204378953, force reloading helped, not sure if something didn't load ok on the first run. 🤷 Cc @filiptronicek

File: components/dashboard/src/settings/SelectIDE.tsx
Comment (gtsiolis) [inline components/dashboard/src/settings/SelectIDE.tsx:183]: **question**: When selecting the latest release, VS Code version remains the same. Is this expected?

**Review by gtsiolis — COMMENTED** [review]
Leaving one last comment, following up from a [relevant discussion](https://gitpod.slack.com/archives/C01KGM9BH54/p1670239613347109?thread_ts=1669993371.377639&cid=C01KGM9BH54) (internal), it should be ok to merge this and create follow-up issues or PRs for the outstanding design details.

Looping in @akosyakov to release the hold break if this sounds ok. 🏓 

Cross-posting for visibility:
> Maybe using a different label variant to not associate these two kinds of information.

> editors

File: components/dashboard/src/settings/SelectIDE.tsx
Comment (akosyakov) [reply components/dashboard/src/settings/SelectIDE.tsx:183]: I don't think we can do it, because JB versions can have or no this prefix already. I think we should go for now as it and then see that we actually will get from JB.

Right now EAP is a released one, so there is no EAP, besides Rider custom handling

Comment (gtsiolis) [reply components/dashboard/src/settings/SelectIDE.tsx:183]: Sounds good. Let's do this! 🤝

Comment (filiptronicek) [reply components/dashboard/src/settings/SelectIDE.tsx:183]: I think here it's because in preview environments, we just use the stable version as Insiders. Should be expected

@iQQBot please correct me if I'm wrong here 😄 🙏

Comment (gtsiolis) [reply components/dashboard/src/settings/SelectIDE.tsx:183]: Thanks, @filiptronicek! OK, let's see how this behaves in production, then. 🎣

**Review by mustard-mh — APPROVED** [review]
Code, UX LGTM.

Tested with preview env, worked like a charm 🎉 

> Seems like this breaks the dashboard UX. ❗

@gtsiolis Maybe related to network issue? you can check your DevTool Console and Network tabs next time you faced it. 🙏

**Comment by akosyakov** [comment]
Can be unhold?

**Comment by filiptronicek** [comment]
Sorry for the confusion, @gtsiolis 🙏. The re-request was triggered because I thought it would unblock merging, it seems that just dismissing your review did the trick (thanks Anton!).

**Comment by gtsiolis** [comment]
Thanks @filiptronicek & @akosyakov!

Opened follow-up issue https://github.com/gitpod-io/gitpod/issues/15194 to keep track of the design nitpicks based on [our discussion](https://gitpod.slack.com/archives/C01KGM9BH54/p1670278365083479?thread_ts=1669993371.377639&cid=C01KGM9BH54) (internal).

**Pull request merged**
